src/kiota/Handlers/KiotaInfoCommandHandler.cs

```diff
@@ -112,15 +117,23 @@ private void ShowLanguagesTable()
         var layout = new StackLayoutView { view };
         console.Append(layout);
     }
-    private void ShowLanguageInformation(GenerationLanguage language, LanguagesInformation informationSource)
+    private void ShowLanguageInformation(GenerationLanguage language, LanguagesInformation informationSource, bool json)
     {
         if (informationSource.TryGetValue(language.ToString(), out var languageInformation))
         {
-            DisplayInfo($"The language {language} is currently in {languageInformation.MaturityLevel} maturity level.",
-                        "After generating code for this language, you need to install the following packages:");
-            foreach (var dependency in languageInformation.Dependencies)
+            if (!json)
+            {
+                DisplayInfo($"The language {language} is currently in {languageInformation.MaturityLevel} maturity level.",
+                            "After generating code for this language, you need to install the following packages:");
+                foreach (var dependency in languageInformation.Dependencies)
+                {
+                    DisplayInfo(string.Format(dependency.Name, dependency.Version));
+                }
+            }
+            else
             {
-                DisplayInfo(string.Format(languageInformation.DependencyInstallCommand, dependency.Name, dependency.Version));
+                var jsonOutput = System.Text.Json.JsonSerializer.Serialize(languageInformation);
```

Comment:
This is going to lead to using reflection.
Can you add a Serialization context (search for JsonSerializable in the code) and pass it to this method please?
(I'm currently doing work to enable trimming, smaller faster binary)

Reply:
This is taking a little longer than expected, looks like I cannot generate `JsonSerializable(typeof(LanguagesInformation))` automatically, I would need to implement it ... next week 🙂

Reply:
found out that it already have a different serialization mechanism built in, @baywet let me know if this works for you.